### ISSUE TITLE
400 Bad Request using Minio S3 backend

### ISSUE THREAD

**Issue by jeffdupont** [restic/restic#2698]
Output of `restic version`
--------------------------
restic 0.9.6 compiled with go1.13.4 on darwin/amd64

How did you run restic exactly?
-------------------------------
restic -r s3:http://192.168.X.X:9000/backups snapshot

AWS_ACCESS_KEY_ID=XXX
AWS_SECRET_ACCESS_KEY=XXX
AWS_REGION=us-east-1

Fatal: unable to open config file: Stat: 400 Bad Request
Is there a repository at the following location?
s3:http://192.168.X.X:9000/backups

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Minio

Expected behavior
-----------------
Prompt for password and list out snapshots for backup that I'm trying to restore

Do you have any idea what may have caused this?
-----------------------------------------------
Restored laptop with newer version of Restic. Prior version used was from before Dec 2017

Do you have an idea how to solve the issue?
-------------------------------------------
No

Additionally I attempted to just create a new bucket on the Minio S3 server and even that fails with a 400 Bad Request. However the error there says "failed: client.BucketExists" and looking at the list of buckets in my minio server, there was no bucket with the name I was attempting to create.

**Comment by rawtaz** [comment]
Please edit/update the issue with the complete output from the run when the problem happens (there was an "Actual behavior" heading you could put it under in the issue template), thanks! :)

**Comment by jeffdupont** [comment]
Sorry, I had posted the output, but apparently I left it inside the comment block. Was wondering why I wasn't seeing it.

**Comment by MichaelEischer** [comment]
Does the log of your minio server contain any helpful error messages?

Otherwise it would be very helpful to use a [debug build (build instructions)](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#reporting-bugs) of restic to get a log of what's happening. The interesting part of the log file would be the part that contains the failing HTTP request & reply. The debug output contains the full HTTP headers, including all access tokens and so on, so make sure to replace these before posting.

**Comment by jeffdupont** [comment]
The debug build and log was an immense help. Before I posted it here scrubbing for the access credentials, I noticed that every request returned a 400. This led me to step through the call with the debugger and found that `be.client.ListObjects` was returning an error that was being ignored that the region was incorrect. Once I corrected the region `AWS_DEFAULT_REGION` to the region used by my minio server I was able to connect and get the password prompt for the backups.
